### PR TITLE
created frontend setup

### PR DESCRIPTION
**Pull request by Mygithub-me**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#Ignore dependencies
+node_modules
+package-lock.json
```
Comment (justuswilhelm) [inline .gitignore:3]: This file shouldn't be ignored and be checked in instead. Also, this file has a missing newline at the end

File: src/components/Welcome.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                </Typography>                
+            </Container>
+            <footer className={classes.footer}>
+                <Container maxWidth="sm">
+                    <Typography variant="body1">Welcome Page!</Typography>                    
+                </Container>
+            </footer>
+        </div>
+    );
+}
```
Comment (justuswilhelm) [inline src/components/Welcome.tsx:46]: Missing newline

File: src/logo.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...21.2 430c-9.3-9.6-18.6-20.3-27.8-32 9 .4 18.2.7 27.5.7 9.4 0 18.7-.2 27.8-.7-9 11.7-18.3 22.4-27.5 32zm-74.4-58.9c-14.2-2.1-27.9-4.7-41-7.9 3.7-12.9 8.3-26.2 13.5-39.5 4.1 8 8.4 16 13.1 24 4.7 8 9.5 15.8 14.4 23.4zM420.7 163c9.3 9.6 18.6 20.3 27.8 32-9-.4-18.2-.7-27.5-.7-9.4 0-18.7.2-27.8.7 9-11.7 18.3-22.4 27.5-32zm-74 58.9c-4.9 7.7-9.8 15.6-14.4 23.7-4.6 8-8.9 16-13 24-5.4-13.4-10-26.8-13.8-39.8 13.1-3.1 26.9-5.8 41.2-7.9zm-90.5 125.2c-35.4-15.1-58.3-34.9-58.3-50.6 0-15.7 22.9-35.6 58.3-50.6 8.6-3.7 18-7 27.7-10.1 5.7 19.6 13.2 40 22.5 60.9-9.2 20.8-16.6 41.1-22.2 60.6-9.9-3.1-19.3-6.5-28-10.2zM310 490c-13.6-7.8-19.5-37.5-14.9-75.7 1.1-9.4 2.9-19.3 5.1-29.4 19.6 4.8 41 8.5 63.5 10.9 13.5 18.5 27.5 35.3 41.6 50-32.6 30.3-63.2 46.9-84 46.9-4.5-.1-8.3-1-11.3-2.7zm237.2-76.2c4.7 38.2-1.1 67.9-14.6 75.8-3 1.8-6.9 2.6-11.5 2.6-20.7 0-51.4-16.5-84-46.6 14-14.7 28-31.4 41.3-49.9 22.6-2.4 44-6.1 63.6-11 2.3 10.1 4.1 19.8 5.2 29.1zm38.5-66.7c-8.6 3.7-18 7-27.7 10.1-5.7-19.6-13.2-40-22.5-60.9 9.2-20.8 16.6-41.1 22.2-60.6 9.9 3.1 19.3 6.5 28.1 10.2 35.4 15.1 58.3 34.9 58.3 50.6-.1 15.7-23 35.6-58.4 50.6zM320.8 78.4z"/><circle cx="420.9" cy="296.5" r="45.7"/><path d="M520.5 78.1z"/></g></svg>
```
Comment (justuswilhelm) [inline src/logo.svg:1]: Could you remove the logo?

File: src/redux/Store.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { createStore, applyMiddleware } from "redux";
+import thunk from "redux-thunk";
+import { composeWithDevTools } from "redux-devtools-extension";
+import { rootReducer } from "./reducers/RootReducer";
+
+const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)))
+
+export default store;
```
Comment (justuswilhelm) [inline src/redux/Store.js:8]: Missing newline

File: src/redux/reducers/RootReducer.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { combineReducers } from "redux";
+
+export const rootReducer = combineReducers({
+
+})
```
Comment (justuswilhelm) [inline src/redux/reducers/RootReducer.js:5]: Missing newline

**Comment by justuswilhelm** [comment]
Please commit package.json.lock
Please change the commit message to "Create new project"

**Review by justuswilhelm — CHANGES_REQUESTED** [review]
Please see the comments for change requests

File: package.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "simpletodo-frontend",
```
Comment (justuswilhelm) [inline package.json:2]: This should say 
```suggestion
  "name": "projectify-frontend",
```

**Comment by justuswilhelm** [comment]
When running npm install, I receive the following warning
```
27 vulnerabilities (16 moderate, 9 high, 2 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force
```

**Comment by justuswilhelm** [comment]
When running `npm start` and opening the page in the browser, I receive the following error:
> Failed to compile
```
./src/components/Welcome.tsx
Module not found: Can't resolve '@material-ui/core/Container' in '/Users/justusperlwitz/projectify-frontend/src/components'
```
>This error occurred during the build time and cannot be dismissed.

**Comment by justuswilhelm** [comment]
I'm still confronted with the following warnings

```
npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

added 2120 packages, and audited 2121 packages in 49s

154 packages are looking for funding
  run `npm fund` for details

27 vulnerabilities (16 moderate, 9 high, 2 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

and when running npm build I get

```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.5.2

Please only submit bug reports when using the officially supported version.
```

I think the problem is with running create-react-app. This comment [here](https://github.com/facebook/create-react-app/issues/11174#issuecomment-986701519) hints to switching to webpack being the solution. Could you evaluate switching to webpack to resolve this?

**Comment by justuswilhelm** [comment]
According to the article I previously linked, switching to next.js might be a good option: https://nextjs.org/docs/migrating/from-create-react-app

**Comment by Mygithub-me** [comment]
> I'm still confronted with the following warnings
> 
> ```
> npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
> npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
> npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
> npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
> npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
> npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
> npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
> npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
> npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
> npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
> npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
> npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
> npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
> npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
> npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
> npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
> 
> added 2120 packages, and audited 2121 packages in 49s
> 
> 154 packages are looking for funding
>   run `npm fund` for details
> 
> 27 vulnerabilities (16 moderate, 9 high, 2 critical)
> 
> To address all issues (including breaking changes), run:
>   npm audit fix --force
> 
> Run `npm audit` for details.
> ```
> 
> and when running npm build I get
> 
> ```
> WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.
> 
> You may find that it works just fine, or you may not.
> 
> SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0
> 
> YOUR TYPESCRIPT VERSION: 4.5.2
> 
> Please only submit bug reports when using the officially supported version.
> ```
> 
> I think the problem is with running create-react-app. This comment [here](https://github.com/facebook/create-react-app/issues/11174#issuecomment-986701519) hints to switching to webpack being the solution. Could you evaluate switching to webpack to resolve this?

**Comment by Mygithub-me** [comment]
> I'm still confronted with the following warnings
> 
> ```
> npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
> npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
> npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
> npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
> npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
> npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
> npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
> npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
> npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
> npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
> npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
> npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
> npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
> npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
> npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
> npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
> 
> added 2120 packages, and audited 2121 packages in 49s
> 
> 154 packages are looking for funding
>   run `npm fund` for details
> 
> 27 vulnerabilities (16 moderate, 9 high, 2 critical)
> 
> To address all issues (including breaking changes), run:
>   npm audit fix --force
> 
> Run `npm audit` for details.
> ```
This I m facing issues in resolving in 'npm audit'

**Comment by Mygithub-me** [comment]
> and when running npm build I get
> 
> ```
> WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.
> 
> You may find that it works just fine, or you may not.
> 
> SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0
> 
> YOUR TYPESCRIPT VERSION: 4.5.2
> 
> Please only submit bug reports when using the officially supported version.
> ```
> 
> I think the problem is with running create-react-app. This comment [here](https://github.com/facebook/create-react-app/issues/11174#issuecomment-986701519) hints to switching to webpack being the solution. Could you evaluate switching to webpack to resolve this?

STILL WORKING ON THIS

File: src/components/Welcome.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import './welcome.css'
+
+export default function Welcome() {
+    return (
+        <div className="container">
+            <div>
+                <h1>Welcome Page!</h1>
+                <br />
+                <p>A welcome page is usually one or more web pages or modal overlays that appear the first time you open an app. The best welcome pages direct a user’s focus to the welcome message, while also orienting them to the product.</p>
```
Comment (justuswilhelm) [inline src/components/Welcome.tsx:10]: Can you change this text to lorem ipsum?

**Review by justuswilhelm — DISMISSED** [review]

**Pull request merged**